modules/hal_rpi_pico/CMakeLists.txt

```
@@ -8,60 +8,66 @@ if(CONFIG_HAS_RPI_PICO)
 	set(common_dir ${ZEPHYR_HAL_RPI_PICO_MODULE_DIR}/src/common)
 	set(boot_stage_dir ${rp2_common_dir}/boot_stage2)
 
-	# Compile the second stage bootloader as a separate executable
-
-	add_executable(boot_stage2)
-
-	target_sources_ifdef(CONFIG_RP2_FLASH_W25Q080 boot_stage2 PRIVATE
-				${boot_stage_dir}/boot2_w25q080.S)
-	target_sources_ifdef(CONFIG_RP2_FLASH_GENERIC_03H boot_stage2 PRIVATE
-				${boot_stage_dir}/boot2_generic_03h.S)
-	target_sources_ifdef(CONFIG_RP2_FLASH_IS25LP080 boot_stage2 PRIVATE
-				${boot_stage_dir}/boot2_is25lp080.S)
-	target_sources_ifdef(CONFIG_RP2_FLASH_W25X10CL boot_stage2 PRIVATE
-				${boot_stage_dir}/boot2_w25x10cl.S)
-	target_sources_ifdef(CONFIG_RP2_FLASH_AT25SF128A boot_stage2 PRIVATE
-				${boot_stage_dir}/boot2_at25sf128a.S)
-
-	target_include_directories(boot_stage2 PUBLIC
-		${CMAKE_CURRENT_LIST_DIR}
-		${boot_stage_dir}/asminclude
-		${rp2_common_dir}/pico_platform/include
-		${rp2040_dir}/hardware_regs/include
-		${common_dir}/pico_base/include
-		${ZEPHYR_BASE}/include
-		)
-
-	target_link_options(boot_stage2 PRIVATE
-		"-nostartfiles"
-		"--specs=nosys.specs"
-		"LINKER:--script=${boot_stage_dir}/boot_stage2.ld"
-		)
-
-	# The second stage bootloader is compiled without kconfig definitions.
-	# Therefore, in order to use toolchain.h, it needs to define CONFIG_ARM.
-	target_compile_definitions(boot_stage2 PRIVATE -DCONFIG_ARM=1)
-
-	# Generates a binary file from the compiled bootloader
-	add_custom_target(bootloader_bin ALL DEPENDS boot_stage2.bin)
-	add_custom_command(OUTPUT boot_stage2.bin
-		DEPENDS boot_stage2
-		COMMAND ${CMAKE_OBJCOPY} -Obinary $<TARGET_FILE:boot_stage2> boot_stage2.bin
-		)
-
-	# Checksums the binary, pads it, and generates an assembly file
-	add_custom_target(boot_stage2_asm ALL DEPENDS boot_stage2.S)
-	add_custom_command(OUTPUT boot_stage2.S
-		DEPENDS boot_stage2.bin
-		COMMAND ${Python3_EXECUTABLE} ${boot_stage_dir}/pad_checksum
-		-s 0xffffffff boot_stage2.bin boot_stage2.S)
+	# The Second Stage Bootloader is only linked to the app that resides
+	# at 0x100. Therefore, only if the app's offset is 0x100, the second
+	# stage bootloader should be compiled.
+	if(${CONFIG_RP2_REQUIRES_SECOND_STAGE_BOOT})
```

remove the de-referencing here:
```suggestion
if(CONFIG_RP2_REQUIRES_SECOND_STAGE_BOOT)
```